instrumentation/cassandra/cassandra-4.4/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_4/CassandraAttributesExtractor.java

```diff
@@ -74,4 +86,39 @@ public void onEnd(
     }
     attributes.put(SemanticAttributes.DB_CASSANDRA_IDEMPOTENCE, idempotent);
   }
+
+  private static void updateServerAddressAndPort(AttributesBuilder attributes, Node coordinator) {
+    EndPoint endPoint = coordinator.getEndPoint();
+    if (endPoint instanceof DefaultEndPoint) {
+      InetSocketAddress address = ((DefaultEndPoint) endPoint).resolve();
+      attributes.put(SemanticAttributes.SERVER_ADDRESS, address.getHostName());
+      attributes.put(SemanticAttributes.SERVER_PORT, address.getPort());
+    } else if (endPoint instanceof SniEndPoint && proxyAddressField != null) {
+      SniEndPoint sniEndPoint = (SniEndPoint) endPoint;
+      Object object = null;
+      try {
+        object = proxyAddressField.get(sniEndPoint);
+      } catch (IllegalAccessException e) {
+        logger.warn(
```

Comment:
`IllegalAccessException` is probably impossible here since `setAccessible` was called. I would have just caught `Exception` here and ignored it.